src/Dropdown.php

```diff
@@ -1612,11 +1615,49 @@ public static function showItemType($types = '', $options = [])
                 'display'             => $params['display'],
                 'rand'                => $params['rand'],
                 'track_changes'       => $params['track_changes'],
+                'init'                => $params['init'],
+                'width'               => $params['width'],
             ]);
         }
         return 0;
     }
 
+    /**
+     * Build dropdown options and check rights if needed
+     *
+     * @param array|string $types
+     * @param bool         $checkright
+     * @return array|string|null
+     */
+    public static function buildDropdownOptions($types, $checkright = false): array|string|null
+    {
+        $options = [];
+
+        if (!is_array($types)) {
+            if ($item = getItemForItemtype($types)) {
+                if ($checkright && !$item->canView()) {
+                    return null;
+                }
+                return $item->getTypeName();
+            }
+
+            return null;
+        }
+
+        foreach ($types as $label => $type) {
+            if (!is_array($type)) {
+                $label = $type;
+            }
+
+            $opts = self::buildDropdownOptions($type, $checkright);
+            if ($opts !== null) {
+                $options[$label] = $opts;
+            }
+        }
+
+        return $options;
+    }
```

Comment:
This method is a bit hard to understand because of the recursion and the multiple parameters/return types.
Not sure how it could be improved thought.